src/Environment.js

```diff
@@ -73,28 +69,35 @@ export default class Environment extends Relay.Environment {
       },
     });
 
-    const networkLayer = this._storeData.getNetworkLayer();
-    const subscriptionObservable = networkLayer.sendSubscription(
-      new SubscriptionRequest(query),
+    const onPayload = (payload) => {
+      updateStoreData(this, subscription.getConfigs(), query, payload);
+    };
+
+    let subscriptionObserver;
+    if (observer) {
+      const definedObserver = observer; // Placate flow.
+      subscriptionObserver = {
+        onNext: (payload) => {
+          onPayload(payload);
+          definedObserver.onNext(payload);
+        },
+        onError: (error) => {
+          definedObserver.onError(error);
+        },
+        onCompleted: () => {
+          definedObserver.onCompleted();
```

Comment:
I could do that, but it's not the signature for observers per RxJS. I figure if the observer we pass into the network layer is guaranteed to have all the handlers, it might be more symmetric to have the same requirement here.